src/brod_sock.erl

```diff
@@ -203,6 +206,38 @@ maybe_upgrade_to_ssl(Sock, _Mod = ssl, SslOpts = [_|_], Timeout) ->
 maybe_upgrade_to_ssl(Sock, _Mod, _SslOpts, _Timeout) ->
   {ok, Sock}.
 
+maybe_sasl_auth(_Sock, _Mod, _ClientId, _Timeout, _SaslOpts = undefined) -> ok;
+maybe_sasl_auth(Sock, Mod, ClientId, Timeout,
```

Comment:
but kafka happily works with PLAINTEXT_SASL listeners. Of course, using plain SASL without SSL in production is wrong on so many levels, but it's useful in debugging. In fact, that's how I did most development. So, maybe we should leave it to ops guys to decide, what do you think?

I'd like to log a warning when *kafka* refuses to accept SASL handshake (with `'UnsupportedSaslMechanism'` or `'IllegalSaslState'` , but I'm not sure where that piece should go. I'm just doing `exit({sasl_auth_error, ...})` for that.